viewing_party/party.py

```diff
@@ -0,0 +1,182 @@
+    if title and genre and rating:
+        new_movie_dict = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+            }
+        return new_movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title): 
+    watch_list = user_data["watchlist"] 
+    watched = user_data["watched"]
+
+    
+    for movie in watch_list:
+        if title in movie["title"]:
+            watch_list.remove(movie)
+            watched.append(movie)
+    return user_data#def create_movie(title,genre,rating):
+    #if title and genre and rating:
+#         movie_dict = {"title": title,
+#         "genre" :genre,
+#         "rating": rating,
+#         }
+#         return movie_dict
+#     else:
+#         return None
+
+
+# def add_to_watched(user_data, movie):
+#     watched = user_data["watched"]
+#     if movie:
+#         watched.append(movie)
+#     else:
+#         watched = []
+#     return user_data
+
+
+
+
+#wave 2
+def get_watched_avg_rating(user_data):
+    # avg_rating = 0.0
+    # rating_sum = 0
+    # for i in range (len(user_data["watched"])):
+    #     rating_sum += user_data["watched"][i]["rating"]
+    #     #calculating
+    # rating_sum/len(user_data["watched"])
+    total_rating = 0
+    if len(user_data["watched"]) == 0:
+        return 0
+    else:
+        for movie in user_data["watched"]:
+            total_rating += movie["rating"]
+    return total_rating/len(user_data["watched"])
+
+
+
+def get_most_watched_genre(user_data):
+    most_recurring_genre= {}
+
+    if len(user_data["watched"]) == 0: 
+        return None 
+
+    for movie_dict in user_data["watched"]:
+        if movie_dict["genre"] in most_recurring_genre:
+            most_recurring_genre[movie_dict["genre"]] +=1
+        else:
+            most_recurring_genre[movie_dict["genre"]] = 1
+
+    max_genre = max(most_recurring_genre, key=most_recurring_genre.get)
+
+
+    return max_genre
+
+
+
+# Wave 3
+
+def get_unique_watched(user_data):
+    friends_list = []
+    user_only_watched = []
+
+    for watched_list in user_data["friends"]:
+        for movie in watched_list["watched"]:
+            friends_list.append(movie["title"])
+
+    for movie in user_data["watched"]:
+        if movie["title"] not in friends_list:
+            user_only_watched.append(movie) 
+
+    return user_only_watched
+
+
+
+def get_friends_unique_watched(user_data):
+    movies_friends_watched = []
+
+    for watched_list in user_data["friends"]:
+        for movie in watched_list["watched"]:
+            if movie not in user_data["watched"] and movie not in movies_friends_watched:
+                movies_friends_watched.append(movie)
+
+    return movies_friends_watched
+
+
+
+
+#Wave 4
+
+def get_available_recs(user_data):
```

Comment:
this is a good implementation. Think about how you could of use one of your previously created functions get_friends_unique_watched(user_data) here to save some lines of code.